packages/next-international/src/app/client/create-i18n-provider-client.tsx

```diff
@@ -3,30 +3,30 @@ import type { Context, ReactNode } from 'react';
 import React, { Suspense, use, useMemo } from 'react';
 import { flattenLocale } from '../../common/flatten-locale';
 import type { LocaleContext } from '../../types';
-
 type I18nProviderProps = Omit<I18nProviderWrapperProps, 'fallback'>;
-
 type I18nProviderWrapperProps = {
   locale: string;
   fallback?: ReactNode;
   children: ReactNode;
 };
-
 export const localesCache = new Map<string, Record<string, unknown>>();
-
 export function createI18nProviderClient<Locale extends BaseLocale>(
   I18nClientContext: Context<LocaleContext<Locale> | null>,
   locales: ImportedLocales,
   fallbackLocale?: Record<string, unknown>,
 ) {
   function I18nProvider({ locale, children }: I18nProviderProps) {
     let clientLocale: any = localesCache.get(locale);
-
     if (!clientLocale) {
-      clientLocale = use(locales[locale as keyof typeof locales]()).default;
+      const newLocale = locales[locale as keyof typeof locales];
+      if (!newLocale) {
+        throw new Error(
```

Comment:
I'm not sure about throwing directly an error - we could log it and try to fallback to another locale instead of crashing the page.

Reply:
Ah, I see. I wasn't too concerned originally since a runtime error was occurring. 

That's nice idea to use fallback locale. However, if the fallback locale isn't provided since it's optional, which locale should we default to ?
 
For now, I've implemented it to call notFound like useCurrentLocale. What do you think? https://github.com/QuiiBz/next-international/pull/252/commits/6435a19173c44b9981a59e8db79f339e5ed53db5